app/api/permissions/routes.ts

```diff
@@ -0,0 +1,55 @@
+import { Application } from 'express';
+import { needsAuthorization } from 'api/auth';
+import { validation } from 'api/utils';
+import { entitiesPermissions } from 'api/permissions/entitiesPermissions';
+import { permissionSchema } from 'shared/types/permissionSchema';
+import { collaborators } from 'api/permissions/collaborators';
+
+export const permissionRoutes = (app: Application) => {
+  app.post(
+    '/api/entities/permissions',
+    needsAuthorization(['admin', 'editor']),
+    validation.validateRequest({
+      properties: {
+        body: {
+          required: ['ids', 'permissions'],
+          properties: {
+            ids: { type: 'array', items: { type: 'string' } },
+            permissions: {
+              type: 'array',
+              items: permissionSchema,
+            },
+          },
+        },
+      },
+    }),
+    async (req, res, _next) => {
+      await entitiesPermissions.setEntitiesPermissions(req.body);
+      res.json(req.body);
+    }
+  );
+
+  app.get('/api/entities/permissions', async (req, res, _next) => {
+    const sharedIds = JSON.parse(req.query.ids);
```

Comment:
we have a parseQuery middleware that handles this in case you wana try use it
https://github.com/huridocs/uwazi/blob/042696c744116b6e1cc52385d997c26f2dce3fde/app/api/utils/parseQueryMiddleware.ts

Reply:
Nice to know, thanks :), it's done